### DISCUSSION TITLE
Need Help with something ?

### DISCUSSION THREAD

**Discussion by deepesh611** [ROFIES-IIITP/Arduino#4] [Q&A]
# Welcome to the ROFIES Q&A Discussion!

Hello ROFIES Members,

We're excited to introduce this new Q&A discussion space where you can ask questions, share knowledge, and collaborate with each other on various robotics projects and topics.

Whether you're dealing with hardware issues, coding challenges, or just looking for advice on a new project idea, this is the place to connect with fellow members and find the support you need.

## How to Get Started:
1. **Ask a Question**: If you're facing any challenges or have doubts, start a new discussion thread with your question. Make sure to provide enough context so others can assist you effectively.
   
2. **Share Your Expertise**: If you have experience or knowledge in a particular area, jump in and help answer questions from other members. Your insights could be incredibly valuable to someone else.
   
3. **Discuss Ideas**: Have a new project idea or want to brainstorm on a concept? Start a discussion to gather feedback and collaborate with others.

## Guidelines:
- Be respectful and supportive of one another.
- Provide clear and detailed responses to help others understand.
- Stay on-topic and use this space to build a collaborative community.

---

Let's work together to make this a helpful and engaging resource for everyone. Happy coding and collaborating!

**Comment by bshe2** [comment]
Hello Team,

I am working on developing an app and need some help as I am new to app development. I am essentially seek guidance and advice regarding the following:
1. Where do I start?
2. What language do I use?
3. How can I code every aspect of the app with just one script?

**Reply by KrishKoria** [reply]
hey, thanks for the question, regarding the answers

1) 
that depends on which platform you decide to go for, initially for starting you can go through youtube tutorials just to get an idea of the various frameworks that exist for app development

Once you decide what you would like to do you can go the course route through udemy or if not, then youtube 

2)
for app development, there are 3 main paths to choose

1. ios development:- using swift which is based on a flavour of c/c++, only go to it if you have mac and an iphone

2. cross platform development:- here you have flutter and react native. flutter is based on dart framework and made by google and react native runs on expo framework and is backed my meta/facebook 

3. native android development :- for this you have java/kotlin with Android studio 




3)
well whichever framework you decide to go for, those are already built for development with a single language, you dont have to worry about it

**Reply by deepesh611** [reply]
> Hello Team,
> 
> I am working on developing an app and need some help as I am new to app development. I am essentially seek guidance and advice regarding the following:
> 
> 1. Where do I start?
> 2. What language do I use?
> 3. How can I code every aspect of the app with just one script?

@bshe2 you can also visit [RoadMap.sh](https://roadmap.sh/) to see how and what you will need along the way. You will get a decent idea of what you should do.
